pyat/at/physics/dynamic_aperture.py

```diff
@@ -0,0 +1,1012 @@
+                    # make sure 0.0 is in the scan
+                    # if d_ is not 0.0:
+                    #    d_.append(0.0);
+
+            # define mesh of points
+            xx, xpxp, yy, ypyp, deltadelta, ctct = np.meshgrid(d_[0],
+                                                               d_[1],
+                                                               d_[2],
+                                                               d_[3],
+                                                               d_[4],
+                                                               d_[5],)
+
+            self.coordinates['x'] = xx.flatten()
+            self.coordinates['xp'] = xpxp.flatten()
+            self.coordinates['y'] = yy.flatten()
+            self.coordinates['yp'] = ypyp.flatten()
+            self.coordinates['delta'] = deltadelta.flatten()
+            self.coordinates['ct'] = ctct.flatten()
+
+        if len(self.coordinates['x']) == 0:
+            raise IOError('grid mode must be grid or radial')
+
+        self.survived = [False]*len(self.coordinates['x'])
+
+        pass
+
+    def test_survived(self, coordinates):
+        """
+        test if a set of particle coordinates survived
+        returns a boolean if the coordinates survived
+        """
+
+        """
+        if delta != 0.0:
+            # recompute orbit
+            self.ring.radiation_off()
+            orbit_delta = self.ring.find_sync_orbit()
+            self.ring.radiation_on()
+
+            test_coord = list(orbit_delta[0])
+        else:
+            test_coord = list(self.orbit[0])
+        """
+
+
+        # nothing to do if no coordinate to test
+        if len(coordinates) == 0:
+            return []
+
+        # if coordinates to test:
+
+        # create 6xN matrix add orbit (with dpp) to each coordinate
+        rin = np.concatenate(([coordinates['x'] + self.orbit[0][0]],
+                              [coordinates['xp'] + self.orbit[0][1]],
+                              [coordinates['y'] + self.orbit[0][2]],
+                              [coordinates['yp'] + self.orbit[0][3]],
+                              [coordinates['delta'] + self.orbit[0][4]],
+                              [coordinates['ct'] + self.orbit[0][5]]),
+                              axis=0)
+
+        # track coordinates for N turns
+        if not self.parallel_computation:
+
+            t, losses = at.atpass(self.ring,
+                                  copy.deepcopy(np.asfortranarray(rin)),
+                                  self.number_of_turns,
+                                  refpts=np.array(np.uint32(0)),  # np.array(np.uint32([len(self.ring)])))
+                                  losses=True)
+
+        else:
+            print('parallel computation')
+
+            # track coordinates
+            t, losses = at.patpass(self.ring,
+                                   copy.deepcopy(rin),
+                                   self.number_of_turns,
```

Comment:
Same here:
```python
            t, losses = at.patpass(self.ring,
                                  rin.copy(),
                                  self.number_of_turns,

```

Reply:
for this I did not need  np.asfortranarray